### PR TITLE
Resolves AirlockVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the AirlockVisualizer into a component/system pair.

### TODO:
- [x] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Pretty sure airlock is a component already

**Comment by TemporalOroboros** [comment]
> Pretty sure airlock is a component already

It's in shared. Do you want me to move the visual data to shared (and thus the server) or give AirlockComponent a client version?

**Comment by jicksaw** [comment]
A bit broken somehow. Tested because I was wondering if it'd fix the airlock bolted/ea lights sometimes not showing.
Peek 2023-02-03 22-26

**Review by AJCM-git — CHANGES_REQUESTED** [review]
There is a lot of code duplication, biggest ones i see are the animations assignations and the state switch, all airlocks have door component but not all doors have airlocks so you can make doorcomponent the primary place to store repeated code

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Ready to reopen. I've fixed the last bug I found (airlocks having active boltlights by default).

**Comment by AJCM-git** [comment]
> Ready to reopen. I've fixed the last bug I found (airlocks having active boltlights by default).

i *think* it reopens automatically as soon as you push to your branch

**Comment by TemporalOroboros** [comment]
I've already pushed to the branch and it hasn't reopened so there may be
something wrong on my end if it's not getting the commits.

File: Content.Client/Doors/DoorSystem.cs [hunk 1]
```diff
@@ -1,28 +1,132 @@
...
+        {
+            if (!_resourceCache.TryGetResource<RSIResource>(SharedSpriteComponent.TextureRoot / baseRsi, out var res))
+            {
+                Logger.Error("Unable to load RSI '{0}'. Trace:\n{1}", baseRsi, Environment.StackTrace);
+            }
+            foreach (ISpriteLayer layer in args.Sprite.AllLayers)
+            {
+                layer.Rsi = res?.RSI;
+            }
+        }
```
Comment (AJCM-git) [inline Content.Client/Doors/DoorSystem.cs:88]: i dont think this is ever used?

File: Content.Client/Doors/DoorSystem.cs [hunk 2]
```diff
@@ -1,28 +1,132 @@
...
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DoorComponent, AppearanceChangeEvent>(OnAppearanceChange);
+    }
+
+    protected override void OnComponentInit(EntityUid uid, DoorComponent comp, ComponentInit args)
+    {
+        comp.OpenSpriteStates = new(new (DoorVisualLayers, string)[1]{(DoorVisualLayers.Base, comp.OpenSpriteState)});
+        comp.ClosedSpriteStates = new(new (DoorVisualLayers, string)[1]{(DoorVisualLayers.Base, comp.ClosedSpriteState)});
```
Comment (AJCM-git) [inline Content.Client/Doors/DoorSystem.cs:28]: can you initialize these more explicitly, its a little hard to read

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Havent tested in-game  but the code is way more cleaner than last time

File: Content.Client/Doors/DoorSystem.cs (hunk 1 above)
Comment (TemporalOroboros) [reply Content.Client/Doors/DoorSystem.cs:88]: Apparently it's used by the airlock painter.

Comment (TemporalOroboros) [reply Content.Client/Doors/DoorSystem.cs:88]: <https://github.com/space-wizards/space-station-14/blob/97a8b64c1d53077bdf62cfc5ce1a069381adb5da/Content.Server/AirlockPainter/AirlockPainterSystem.cs#L48>

File: Content.Client/Doors/DoorSystem.cs (hunk 2 above)
Comment (TemporalOroboros) [reply Content.Client/Doors/DoorSystem.cs]: Done.

File: Content.Client/Doors/DoorSystem.cs (hunk 1 above)
Comment (AJCM-git) [reply Content.Client/Doors/DoorSystem.cs:91]: huh

**Review by metalgearsloth — COMMENTED** [review]
Formatting is a bit over the shop.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**